third_party/terraform/utils/utils.go.erb

```diff
@@ -477,3 +477,16 @@ func absoluteDomainSuppress(k, old, new string, _ *schema.ResourceData) bool {
 	return old == new
 }
 <% end -%>
+
+func getInterconnectAttachmentLink(config *Config, project, region, ic string) (string, error) {
```

Comment:
I think you're using `getVpnTunnelLink` as a template here but I don't think it's setting good patterns. 

In general I would lean towards encouraging people to use [interconnect attachments `self_link`](https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_compute_interconnect_attachment.go#L148) because it contains enough of a path to get the resource without needing to infer the region. 

However, given that vpnTunnel supports both `name` and `self_link` I think it's more consistent for both to behave the same so see comments below.